src/__tests__/hooks.test.js

```diff
@@ -0,0 +1,930 @@
+      key: { x: 3, y: 3, z: 2 },
+    });
+  });
+
+  it('will call dispatchOnMount as a function', () => {
+    const testDispatchOnMount = { test: true };
+    const dispatchOnMount = jest.fn(() => ({ dom: true }));
+
+    const TestComponent = () => {
+      useTracking(testDispatchOnMount, { dispatch, dispatchOnMount }); // TODO: potential recipe here is if a component does not render children, it doesn't need to include <Track /> in the tree!
+      return null;
+    };
+
+    mount(<TestComponent />);
+
+    expect(dispatchOnMount).toHaveBeenCalledWith(testDispatchOnMount);
+    expect(dispatch).toHaveBeenCalledWith({ dom: true, test: true });
+  });
+
+  it('will dispatch a pageview event on mount on class component', () => {
+    const App = ({ children }) => {
+      const { Track } = useTracking(
+        { topLevel: true },
+        {
+          dispatch,
+          process: data => {
+            if (data.page) {
+              return { event: 'pageView' };
+            }
+            return null;
+          },
+        }
+      );
+
+      return <Track>{children}</Track>;
+    };
+
+    const Page = () => {
+      useTracking({ page: 'Page' }); // TODO: Does this pass? It doesn't render children so we should still get all the proper tracking context dispatched on mount according to "process" fn above
+      return <div>Page</div>;
+    };
+
+    mount(
+      <App>
+        <Page />
+      </App>
+    );
+
+    expect(dispatch).toHaveBeenCalledWith({
+      topLevel: true,
+      event: 'pageView',
+      page: 'Page',
+    });
+  });
+
+  it('will dispatch a pageview event on mount on functional component', () => {
+    // TODO: Using purely hooks API, this test is redundant with above, but we will keep it for parity
+    const App = ({ children }) => {
+      const { Track } = useTracking(
+        { topLevel: true },
+        {
+          dispatch,
+          process: data => {
+            if (data.page) {
+              return { event: 'pageView' };
+            }
+            return null;
+          },
+        }
+      );
+
+      return <Track>{children}</Track>;
+    };
+
+    const Page = () => {
+      useTracking({ page: 'Page' });
+      return <div>Page</div>;
+    };
+
+    mount(
+      <App>
+        <Page />
+      </App>
+    );
+
+    expect(dispatch).toHaveBeenCalledWith({
+      topLevel: true,
+      event: 'pageView',
+      page: 'Page',
+    });
+  });
+
+  it("should not dispatch a pageview event on mount if there's no page property on tracking object", () => {
+    const App = ({ children }) => {
+      const { Track } = useTracking(
+        { topLevel: true },
+        {
+          dispatch,
+          process: data => {
+            if (data.page) {
+              return { event: 'pageView' };
+            }
+            return null;
+          },
+        }
+      );
+
+      return <Track>{children}</Track>;
+    };
+
+    const Page = () => {
+      useTracking({ page: 'Page' });
```

Comment:
How is this different from the 2 tests above this one, where it does dispatch a pageview event?

Reply:
Ah good catch, I think this was a bad copy pasta, the process function should no-op to match up with the [corresponding e2e test](https://github.com/nytimes/react-tracking/blob/919108b8874b5121b4b20927fa666737e2484ac5/src/__tests__/e2e.test.js#L239).